File: androidx/SDKSampleJava/app/src/main/java/com/gyant/sdksamplejava/DisplayGyantViewWithPushTokenActivity.java
package com.gyant.sdksamplejava;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.FrameLayout;

import com.gyant.chat_sdk.GyantChat;
import com.gyant.chat_sdk.GyantOnPushTokenListener;
import com.gyant.chat_sdk.CompletionHandler;
import com.gyant.chat_sdk.GyantView;


public class DisplayGyantViewWithPushTokenActivity extends AppCompatActivity implements GyantOnPushTokenListener{
    GyantView gyantView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_display_gyant_view);

        GyantChat gyantChat = GyantChat.getInstance()
                .clientId("client_id")
                .patientId("patient_id")
                .setOnPushTokenListener(this)
                .isDev(true)
                .start();

        this.gyantView = gyantChat.createView(this, getLifecycle());

        FrameLayout frameLayout = (FrameLayout) findViewById(R.id.frame_layout);
        frameLayout.addView(gyantView);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        this.gyantView.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    @Override
    public void getPushToken(CompletionHandler<String> callback){
        callback.onComplete("token123");
    }
}

File: androidx/SDKSampleKotlin/app/src/main/java/com/gyant/sdksamplekotlin/DisplayGyantViewWithPushTokenActivity.kt
package com.gyant.sdksamplekotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.FrameLayout

import com.gyant.chat_sdk.CompletionHandler
import com.gyant.chat_sdk.GyantChat
import com.gyant.chat_sdk.GyantView
import com.gyant.chat_sdk.GyantOnPushTokenListener


class DisplayGyantViewWithPushTokenActivity : AppCompatActivity(), GyantOnPushTokenListener {
    internal lateinit var gyantView: GyantView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_display_gyant_view)

        val gyantChat = GyantChat.getInstance()
            .clientId("client_id")
            .patientId("patient_id")
            .setOnPushTokenListener(this)
            .isDev(true)
            .start()

        this.gyantView = gyantChat.createView(this, lifecycle)

        val frameLayout = findViewById<View>(R.id.frame_layout) as FrameLayout
        frameLayout.addView(gyantView)
    }


    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String>,
        grantResults: IntArray
    ) {
        this.gyantView.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }


    override fun getPushToken(callback: CompletionHandler<String>) {
        callback.onComplete("token123")
    }
}